packages/peregrine/lib/talons/CartPage/ProductListing/useProduct.js

```diff
@@ -3,6 +3,23 @@ import { useMutation } from '@apollo/react-hooks';
 import { useAppContext } from '@magento/peregrine/lib/context/app';
 import { useCartContext } from '@magento/peregrine/lib/context/cart';
 
+/**
+ * This talon contains logic for a product component used in a product listing component.
+ * It performs effects and returns prop data for that component.
```

Comment:
Now this one has multiple effects, and we probably still want a concise description; open to ideas on how to document the behavior of a talon.

1. While either mutation is in flight we sync that with parent cart state
2. When global app state indicates the drawer has been closed, we reset the active item back to `null`. This allows components to unmount between renders to reset form state.

Reply:
Description updated with effects description